mlir/lib/Dialect/DLTI/DLTI.cpp

```diff
@@ -562,6 +565,19 @@ dlti::query(Operation *op, ArrayRef<DataLayoutEntryKey> keys, bool emitError) {
   return currentAttr;
 }
 
+FailureOr<Attribute> dlti::query(Operation *op, ArrayRef<StringRef> keys,
+                                 bool emitError) {
+  if (!op)
+    return failure();
+
+  MLIRContext *ctx = op->getContext();
+  SmallVector<DataLayoutEntryKey> entryKeys;
```

Comment:
Nit: `entryKeys` will become the size of `keys`, so we can allocate the space for it upfront (even though the length of `keys` is likely so small that it will fit in the stack-allocated space any way).